bin/cli.rs

```diff
@@ -0,0 +1,126 @@
+#![allow(missing_docs)]
+use std::{
+    collections::BTreeMap,
+    fs::{create_dir_all, File},
+    io::BufReader,
+    num::NonZeroUsize,
+    thread,
+};
+
+use alloy_consensus::constants::KECCAK_EMPTY;
+use alloy_primitives::{Address, B256};
+use alloy_provider::{Provider, ProviderBuilder};
+use alloy_rpc_types::{BlockId, BlockTransactionsKind};
+use clap::{Parser, Subcommand};
+use pevm::{
+    chain::{PevmChain, PevmEthereum},
+    EvmAccount, EvmCode, Pevm, RpcStorage,
+};
+use reqwest::Url;
+use tokio::runtime;
+
+#[derive(Parser, Debug)]
+pub struct Cli {
+    #[clap(subcommand)]
+    pub subcmd: Commands,
+}
+
+#[derive(Subcommand, Debug)]
+pub enum Commands {
+    /// Snapshot a block to disk.
+    Snapshot {
+        /// The ID of the block.
+        block_id: BlockId,
+        /// The RPC url to connect to.
+        rpc_url: String,
+        /// The output directory to write the snapshot to.
+        output_dir: String,
+        /// Path to bytecodes.
+        bytecode_path: Option<String>,
+    },
+}
+
+pub fn main() {
+    let args = Cli::parse();
+
+    match args.subcmd {
+        Commands::Snapshot {
+            block_id,
+            rpc_url,
+            output_dir,
+            bytecode_path,
+        } => {
+            // Define provider.
+            let provider = ProviderBuilder::new().on_http(
+                Url::parse(&rpc_url)
+                    .unwrap_or_else(|_| panic!("Invalid RPC URL supplied: {rpc_url}")),
```

Comment:
We can shorten these `.unwrap_or_else` with `.expect`?

Reply:
Yeah, this was my original thought. However clippy doesn't like that format with dynamic arguments:
```
.expect(format!("No block found for ID: {:?}", block_id).as_str());
   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `unwrap_or_else(|| panic!("No block found for ID: {:?}", block_id))`
 ```

Reply:
Ah, no worries then, let's just follow `clippy` for consistency 🙏.